### PR TITLE
JSON Schema: Provide support for discovery service

### PR DESCRIPTION
**Pull request by ccaraman**
-Updates documentation to refer to schema validation
-Adds schema support for SDS and CDS.

### CODE REVIEW

**Comment by ccaraman** [comment]
@lyft/network-team

File: docs/configuration/configuration.rst [hunk 1]
```diff
@@ -3,6 +3,9 @@
 Configuration reference
 =======================
 
+Configurations are validated against a JSON schema. The schema is separated into several parts and
```
Comment (mattklein123) [inline docs/configuration/configuration.rst]: I think I would probably move this into the overview page.

File: test/common/upstream/sds_test.cc [hunk 2]
```diff
@@ -304,4 +304,27 @@ TEST_F(SdsTest, HealthChecker) {
...
+  )EOF";
+
+  Http::MessagePtr message(new Http::ResponseMessageImpl(
+      Http::HeaderMapPtr{new Http::TestHeaderMapImpl{{":status", "200"}}}));
+  message->body(Buffer::InstancePtr{new Buffer::OwnedImpl(bad_response_json)});
+
+  EXPECT_CALL(*timer_, enableTimer(_));
+  callbacks_->onSuccess(std::move(message));
+
+  EXPECT_CALL(*timer_, enableTimer(_));
```
Comment (mattklein123) [inline test/common/upstream/sds_test.cc]: You should be calling timer_->callback_() before this, but I would just delete this second case. It's not relevant to what you are testing.

File: docs/configuration/overview/overview.rst [hunk 3]
```diff
@@ -3,7 +3,8 @@
 Overview
 ========
 
-The Envoy configuration format is written in JSON. The main configuration for the server is
+The Envoy configuration format is written in JSON and is validated against a JSON schema. The
+schema can be found here :repo:`source/common/json/config_schemas.cc`. The main configuration for the server is
```
Comment (mattklein123) [inline docs/configuration/overview/overview.rst]: either make 'here' the actual link, or "can be found in the ... file"

File: docs/configuration/configuration.rst (hunk 1 above)
Comment (ccaraman) [inline docs/configuration/configuration.rst]: Will remove this entry.

**Pull request merged**